libkode/printer.cpp

```diff
@@ -956,16 +951,47 @@ void Printer::printImplementation( const File &file, bool createHeaderInclude )
     filename.prepend( d->mOutputDirectory + '/' );
 
   QFile implementation( filename );
-  if ( !implementation.open( QIODevice::WriteOnly ) ) {
-    qWarning( "Can't open '%s' for writing.", qPrintable( filename ) );
-    return;
+  printCodeIntoFile(out, &implementation);
+}
+
+void Printer::printCodeIntoFile(const Code &code, QFile *file)
+{
+  QString outText = code.text();
```

Comment:
prepend "const"

Reply:
The QTextStream only has a non const constructor for QStrings:

`    explicit QTextStream(QString *string, QIODevice::OpenMode openMode = QIODevice::ReadWrite);

which utilized here:

https://github.com/martonmiklos/kode/blob/skip_owerwrite_if_output_identical/libkode/printer.cpp#L967

So utilizing const is not possible here.

Reply:
I see. This would require using the QTextStream(const QByteArray&) constructor instead, which means converting the QString to a QByteArray using the same encoding as the one used in the file. The usual way for files is to use toLocal8Bit() in general, although for code we want it all utf-8, even on a non-utf8-Windows.

So I would do that here, and then make sure the fileStream is set to utf-8 too.

This isn't just about being able to add a const, it's about making sure we're only reading from the string, avoiding  hidden detaches, and making sure the right encoding is used everywhere (too implicit right now).